js/pages/incidence-rates/ir-manager.js

```diff
@@ -275,6 +275,9 @@ define([
 			const { analysisId } = params;
 			if (analysisId && parseInt(analysisId) !== (this.selectedAnalysis() && this.selectedAnalysis().id())) {
 				this.onAnalysisSelected();
+			} else if (this.selectedAnalysis() && this.selectedAnalysis().id()) {
+				this.pollForInfo();
+				this.pollTimeout = PollService.add(() => this.pollForInfo({ silently: true }), 10000);
```

Comment:
These two lines:
```
this.pollForInfo();
this.pollTimeout = PollService.add(() => this.pollForInfo({ silently: true }), 10000);
```
already exist in `onAnalysisSelected`.

Wouldn't it be better to extract those into a separate function? Or even introduce a new flag in `PollService.add` which would kick off the provided function immediately (so encapsulate the behaviour)

Reply:
@pavgra, agree and i also noticed that in PollService at first we call fn and then wait for another call, so it's no need to call for ex. this.pollForInfo() before adding it to PollService.